src/panels/config/integrations/integration-panels/zwave_js/zwave_js-config-dashboard.ts

```diff
@@ -627,6 +607,41 @@ class ZWaveJSConfigDashboard extends SubscribeMixin(LitElement) {
     showOptionsFlowDialog(this, configEntry!);
   }
 
+  private _openInclusionDialog(dsk?: string) {
+    if (!this._dialogOpen) {
+      // Unsubscribe from S2 inclusion before opening dialog
+      if (this._s2InclusionUnsubscribe) {
+        this._s2InclusionUnsubscribe();
+        this._s2InclusionUnsubscribe = undefined;
+      }
+
+      showZWaveJSAddNodeDialog(this, {
+        entry_id: this.configEntryId!,
+        dsk,
+        onStop: this._handleInclusionDialogClosed,
+      });
+      this._dialogOpen = true;
+    }
+  }
+
+  private _handleInclusionDialogClosed = () => {
+    // refresh the data after the dialog is closed. add a small delay for the inclusion state to update
+    setTimeout(() => this._fetchData(), 100);
+    this._dialogOpen = false;
+    this._subscribeS2Inclusion();
+  };
+
+  private async _subscribeS2Inclusion() {
+    this._s2InclusionUnsubscribe = await subscribeS2Inclusion(
```

Comment:
It is better not to `await` here, but when we want to unsubscribe, as now `_s2InclusionUnsubscribe` can still be undefined when we open the dialog while it is subscribed.